Visual Studio 2017 will crash with this code.

**15.7.2**: 

**Steps to Reproduce**:

1. write this code:
```
      foreach(_ int "")
            {
            }
```
2. move mouse to _ ,wait about 2 seconds

**Expected Behavior**: nohting

**Actual Behavior**:Visual Studio 2017 will crash

I am unable to repro this in master or the latest VS preview 2.0 candidate. Here is a test I am adding:

``` c#
        [Fact, WorkItem(27104, "https://github.com/dotnet/roslyn/issues/27104")]
        public void BadDiscardInForeachLoop_01()
        {
            var source = @"
class C
{
    void M()
    {
        foreach(_ in """")
        {
        }
    }
}";
            var compilation = CreateCompilation(source);
            compilation.VerifyDiagnostics(
                // (6,17): error CS8186: A foreach loop must declare its iteration variables.
                //         foreach(_ in "")
                Diagnostic(ErrorCode.ERR_MustDeclareForeachIteration, "_").WithLocation(6, 17)
                );
            var tree = compilation.SyntaxTrees[0];
            var variable = tree.GetRoot().DescendantNodes().OfType<ForEachVariableStatementSyntax>().Single().Variable;
            var model = compilation.GetSemanticModel(tree);
            var symbol = model.GetSymbolInfo(variable).Symbol;
            Verify(
                symbol.ToMinimalDisplayParts(model, variable.SpanStart),
                "var _",
                SymbolDisplayPartKind.ErrorTypeName, // var
                SymbolDisplayPartKind.Space,
                SymbolDisplayPartKind.Punctuation); // _
        }
```

--- a/src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs
@@ -6580,5 +6580,36 @@ class X
                 SymbolDisplayPartKind.Space,
                 SymbolDisplayPartKind.Keyword);
         }
+
+        [Fact, WorkItem(27104, "https://github.com/dotnet/roslyn/issues/27104")]
+        public void BadDiscardInForeachLoop_01()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        foreach(_ in """")
+        {
+        }
+    }
+}";
+            var compilation = CreateCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (6,17): error CS8186: A foreach loop must declare its iteration variables.
+                //         foreach(_ in "")
+                Diagnostic(ErrorCode.ERR_MustDeclareForeachIteration, "_").WithLocation(6, 17)
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var variable = tree.GetRoot().DescendantNodes().OfType<ForEachVariableStatementSyntax>().Single().Variable;
+            var model = compilation.GetSemanticModel(tree);
+            var symbol = model.GetSymbolInfo(variable).Symbol;
+            Verify(
+                symbol.ToMinimalDisplayParts(model, variable.SpanStart),
+                "var _",
+                SymbolDisplayPartKind.ErrorTypeName, // var
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.Punctuation); // _
+        }
     }
 }

